packages/dm-core-plugins/src/generic-list/GenericTablePlugin.tsx

```diff
@@ -0,0 +1,190 @@
+import { delete_to_trash, library_add } from '@equinor/eds-icons'
+
+type TGenericTablePlugin = {
+  editMode: boolean
+  columns: string[]
+  showDelete: boolean
+}
+const defaultConfig: TGenericTablePlugin = {
+  editMode: true,
+  columns: ['name', 'type'],
+  showDelete: true,
+}
+
+export const GenericTablePlugin = (
+  props: IUIPlugin & { config?: TGenericTablePlugin }
+) => {
+  const { idReference, config } = props
+  const internalConfig = { ...defaultConfig, ...config }
+  const [document, loading, _, error] = useDocument<TGenericObject[]>(
+    idReference,
+    2
+  )
+  const [items, setItems] = useState<{ [key: string]: TGenericObject }>({})
+  const [isSaveLoading, setIsSaveLoading] = useState<boolean>(false)
+  const { token } = useContext(AuthContext)
+  const dmssAPI = new DmssAPI(token)
+
+  useEffect(() => {
+    if (loading || !document) return
+    // Need to generate a uuid for each item in the list to be used for reacts "key" property
+    const itemsWithIds = Object.fromEntries(
+      Object.values(document).map((value: any) => [crypto.randomUUID(), value])
+    )
+    setItems(itemsWithIds)
+  }, [document, loading])
+
+  const deleteItem = (reference: string, key: string) => {
+    dmssAPI.documentRemove({ idReference: reference }).then(() => {
+      delete items[key]
+      setItems({ ...items })
+    })
+  }
+  const addItem = (reference: string, type: string) => {
+    dmssAPI
+      // TODO: Get type from parent blueprint, be able to select specialised type
+      .instantiateEntity({
+        entity: { type: items[Object.keys(items)[0]].type },
+      })
+      .then((newEntity: AxiosResponse<object, TGenericObject>) => {
+        dmssAPI
+          .documentAdd({ absoluteRef: reference, body: newEntity.data })
+          .then(() =>
+            setItems({
+              ...items,
+              [crypto.randomUUID()]: { ...newEntity.data },
+            })
+          )
+      })
+  }
+
+  const updateItem = (
+    key: string,
+    attribute: string,
+    newValue: string | number | boolean
+  ) => {
+    items[key][attribute] = newValue
+    setItems({ ...items })
+  }
+
+  const saveTable = () => {
+    setIsSaveLoading(true)
+    dmssAPI
+      .documentUpdate({
+        idReference: idReference,
+        data: JSON.stringify(Object.values(items)),
+      })
+      .catch((error: AxiosError<ErrorResponse>) =>
+        alert(JSON.stringify(error.response?.data))
+      )
+      .finally(() => setIsSaveLoading(false))
+  }
+
+  if (loading) return <Loading />
+  if (error) throw new Error(JSON.stringify(error, null, 2))
+  if (!items) return <Loading />
+
+  return (
+    <>
+      <div
+        style={{ display: 'flex', flexDirection: 'row-reverse', width: '100%' }}
+      >
+        <Button
+          variant="ghost_icon"
+          onClick={() => addItem(idReference, 'type')}
+        >
+          <Icon data={library_add} title="Append" />
+        </Button>
+      </div>
```

Comment:
Is the div necessary?

Reply:
Any suggestions on how to move the button to far right without this wrapper?

Reply:
style={{ float: 'right' }} works nicely

Reply:
Alternatively 
![image](https://user-images.githubusercontent.com/64828956/232428757-57104a38-1b26-4291-aef0-64dc57dd5453.png)